spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelFactory.java

```diff
@@ -25,5 +25,7 @@
  */
 public interface ChannelFactory {
```

Comment:
While we can use `MessageChannel` as the target type instead of DirectChannel in the case above, I went ahead and parameterize the `ChannelFactory` interface.